File: leetcode/lc_069.c
//
// LeetCode
// 69. Sqrt(x)
// https://leetcode.com/problems/sqrtx/
//


int mySqrt(int x) {
    int low = 1;
    int high = x;
    int guess;
    
    while (low <= high) {
        guess = low + (high - low) / 2;
        
        if (guess < x / guess)
            low = guess + 1;
        else if (guess  > x / guess)
            high = guess - 1;
        else
            return guess;
    }
    
    return high;
}

File: leetcode-swift/lc_069.swift
//
// LeetCode
// 69. Sqrt(x)
// https://leetcode.com/problems/sqrtx/
//


class Solution {
    func mySqrt(_ x: Int) -> Int {
        var low = 1
        var high = x
        var guess: Int
        
        while low <= high {
            guess = low + (high - low) / 2
            
            if guess < x / guess {
                low = guess + 1
            } else if guess > x / guess {
                high = guess - 1
            } else {
                return guess
            }
        }
        
        return high
    }
}
